backend/backend/Controllers/UsersController.cs

```diff
@@ -57,20 +57,78 @@ public ActionResult<User> GetCurrentUser()
                 return NotFound();
             }
 
-            return user;
+            UserDto userDto = new UserDto()
+            {
+                Id = user.Id,
+                FirstName = user.Firstname,
+                LastName = user.Lastname,
+                Email = user.Email
+            };
+
+            return userDto;
         }
 
         [HttpPost, Route("authentication"), AllowAnonymous]
-        public ActionResult<string> Authenticate([FromBody]UserCredentials userCredentials)
+        public ActionResult<string> Authenticate(string email, string password)
         {
-            var token = _jwtAuthentication.Authenticate(userCredentials.Email, userCredentials.Password);
+            if(String.IsNullOrWhiteSpace(email) || String.IsNullOrWhiteSpace(password))
+            {
+                return BadRequest();
+            }
 
-            if(token is null)
+            var token = _jwtAuthentication.Authenticate(email, password);
+
+            if (token is null)
             {
                 return Unauthorized();
             }
 
             return Ok(token);
         }
+
+        [HttpPost, Route("register"), AllowAnonymous]
+        public ActionResult Register([FromBody] RegistrationRequest registrationRequest)
+        {
+            if (String.IsNullOrWhiteSpace(registrationRequest.Email) || !RegexValidation.IsEmailValid(registrationRequest.Email)) return BadRequest();
+            if (String.IsNullOrWhiteSpace(registrationRequest.FirstName) || String.IsNullOrWhiteSpace(registrationRequest.LastName)
+                || !RegexValidation.IsNameValid(registrationRequest.FirstName) || !RegexValidation.IsNameValid(registrationRequest.LastName)) return BadRequest();
+            if (String.IsNullOrWhiteSpace(registrationRequest.Password) || !RegexValidation.IsPasswordValid(registrationRequest.Password)) return BadRequest();
+            if (_db.Users.FirstOrDefault(user => user.Email == registrationRequest.Email) != null) return Conflict();
+
+            User user = new User(registrationRequest.Password)
+            {
+                Firstname = registrationRequest.FirstName,
+                Lastname = registrationRequest.LastName,
+                Email = registrationRequest.Email
+            };
+
+            _db.Users.Add(user);
+
+            try
+            {
+                string endpoint = "http://localhost:3000/verify/" + user.Id;
+                _emailService.SendVerificationEmail(user.Firstname, user.Email, endpoint);
+                _db.SaveChanges();
+            }
+            catch
+            {
+                return StatusCode(500);
+            }
+
+            return Ok();
+        }
+
+        [HttpPost, Route("verify"), AllowAnonymous]
+        public ActionResult Verify(Guid id)
+        {
+            if (id == Guid.Empty) return NotFound();
```

Comment:
Bro, you mixed these up. If Guid is empty then it's bad request and if the user is null then it's NotFound

Reply:
True that my friend